### PR TITLE
Generalize "creation/archive" of maf database

### PR DESCRIPTION
**Pull request by thomasyu888**
The creation of the "maf" database is something specific to GENIE, but the creation of a new file format table is not.  Restructured code such that the function:

1.  Creates a new fileformat table based on an old one
2. Updates the database mapping table
3. moves the old fileformat table to a new location with a new name (Append ARCHIVED to the old name)

### CODE REVIEW

**Comment by thomasyu888** [comment]
Don't pull in yet, need to fix test

File: genie/process_functions.py [hunk 1]
```diff
@@ -1041,3 +1041,112 @@ def get_gdc_data_dictionary(filetype):
...
+        columns: Columns of Table
+        annotations: Dictionary of annotations to add
+
+    Returns:
+        Schema
+    """
+    schema = synapseclient.Schema(name=table_name,
+                                  columns=columns,
+                                  parent=parentid)
+    schema.annotations = annotations
```
Comment (kdaily) [inline genie/process_functions.py]: move to constructor

File: genie/process_functions.py [hunk 2]
```diff
@@ -1041,3 +1041,112 @@ def get_gdc_data_dictionary(filetype):
...
+    # mapping table
+    to_update_row = database_synid_mappingdf[fileformat_ind]
+
+    # Update this synid later (This synid needs to not be hardcoded)
+    updated_table = syn.store(synapseclient.Table(database_mapping_synid,
+                                                  to_update_row))
+    return updated_table
+
+
+def _move_entity(syn, ent, parentid, name=None):
```
Comment (kdaily) [inline genie/process_functions.py:1097]: TODO deprecate once some jira for move function is out of the cli into the client

File: genie/process_functions.py [hunk 3]
```diff
@@ -1041,3 +1041,112 @@ def get_gdc_data_dictionary(filetype):
...
+    """Archives an older table
```
Comment (kdaily) [inline genie/process_functions.py]: move, not just for tables

File: genie/process_functions.py [hunk 4]
```diff
@@ -1041,3 +1041,112 @@ def get_gdc_data_dictionary(filetype):
...
+                               parentid=projectid,
+                               annotations=olddb_ent.annotations)
+
+    updated_table = _update_database_mapping(syn, database_synid_mappingdf,
+                                             database_synid_mapping.tableId,
+                                             file_format, newdb_ent.id)
+    # Automatically rename the archived entity with ARCHIVED
+    # This will attempt to resolve any issues if the table already exists at
+    # location
+    new_table_name = "ARCHIVED {}".format(newdb_ent.name)
```
Comment (kdaily) [inline genie/process_functions.py]: add timestamp so know when it was archived

File: bin/create_maf_database.py [hunk 5]
```diff
@@ -1,24 +1,38 @@
...
+        syn, "main", databaseToSynIdMappingDf=database_mappingdf)
+
+    new_tables = process_functions.create_new_fileformat_table(syn, database_mapping,
+                                                               "vcf2maf",
+                                                               table_name,
+                                                               projectid,
+                                                               "syn7208886")
+
+    # Remove can download permissions from project GENIE team
+    syn.setPermissions(new_tables['newdb_ent'].id, 3326313, [])
```
Comment (kdaily) [inline bin/create_maf_database.py]: this wouldn't be necessary if just move to a projec that isn't viewable by GENIE peeps.

**Review by kdaily — CHANGES_REQUESTED** [review]

File: genie/process_functions.py (hunk 2 above)
Comment (thomasyu888) [reply genie/process_functions.py:1097]: Actually the `move` functionality is already part of the `develop` branch.  https://github.com/Sage-Bionetworks/synapsePythonClient/pull/550/files, but it doesn't have the option of renaming.

**Comment by kdaily** [comment]
💯

**Pull request closed without merging**